### PR TITLE
Adding HTTP(S) tests to the cache (SOFTWARE-3936)

### PR DESCRIPTION
**Pull request by efajardo**

### CODE REVIEW

File: osgtest/tests/test_155_stashcache.py [hunk 1]
```diff
@@ -77,10 +79,15 @@
 """
 
 CACHE_AUTHFILE_PATH = PARAMS["StashCacheAuthfile"]
-CACHE_AUTHFILE_CONTENTS = "u * / rl\n"
+CACHE_AUTHFILE_CONTENTS = "u b64f6609.0 /osgtest/PROTECTED rl\n"
```
Comment (matyasselmeci) [inline osgtest/tests/test_155_stashcache.py]: What's "b64f6609.0"?

File: osgtest/tests/test_460_stashcache.py [hunk 2]
```diff
@@ -100,3 +107,53 @@ def test_05_stashcp(self):
...
+        path = os.path.join(getcfg("OriginAuthExport"), name)
+        dest_file = '/tmp/testfileFromOriginAuth'
+        os.environ["XrdSecGSISRVNAMES"] = "*"
+        result, _, _ = core.check_system(["xrdcp", "-d1", '-f', 
+                                          "root://localhost:%d/%s" % (getcfg("OriginAuthXrootPort"), path),
+                                          dest_file],
+                                         "Checking xrootd copy from authenticated origin", user=True)
+        origin_file = os.path.join(getcfg("OriginRootdir"), getcfg("OriginAuthExport").lstrip("/"), name)
+        chechskum_match = files.checksum_files_match(origin_file, dest_file)
+        self.assert_(chechskum_match, 'Origin and download file have same contents')
```
Comment (matyasselmeci) [inline osgtest/tests/test_460_stashcache.py]:
```suggestion
        checksum_match = files.checksum_files_match(origin_file, dest_file)
        self.assert_(checksum_match, 'Origin and directly downloaded file have the same contents')
```

File: osgtest/tests/test_460_stashcache.py [hunk 3]
```diff
@@ -100,3 +107,53 @@ def test_05_stashcp(self):
...
+        path = os.path.join(getcfg("OriginAuthExport"), name)
+        os.environ["XrdSecGSISRVNAMES"] = "*"
+        dest_file = '/tmp/testfileXrootdFromAuthCache'
+        result, _, _ = \
+            core.check_system(["xrdcp", "-d1","-f",
+                               "root://%s:%d/%s" % (core.get_hostname(),getcfg("CacheHTTPSPort"), path),
+                               dest_file], "Checking xrootd copy from Authenticated cache", user=True)
+        origin_file = os.path.join(getcfg("OriginRootdir"), getcfg("OriginAuthExport").lstrip("/"), name)
+        chechskum_match = files.checksum_files_match(origin_file, dest_file)
+        self.assert_(chechskum_match, 'Cache and download file have same contents')
```
Comment (matyasselmeci) [inline osgtest/tests/test_460_stashcache.py]:
```suggestion
        checksum_match = files.checksum_files_match(origin_file, dest_file)
        self.assert_(checksum_match, 'Origin and file downloaded via cache have the same contents')
```

File: osgtest/tests/test_460_stashcache.py [hunk 4]
```diff
@@ -100,3 +107,53 @@ def test_05_stashcp(self):
...
+        userkey = '/tmp/x509up_u%d' % uid
+        result, _, _ = \
+            core.check_system(["gfal-copy", "-vf",
+                               "--cert", usercert, "--key", userkey,
+                               "https://%s:%d%s" % (core.get_hostname(),getcfg("CacheHTTPSPort"), path),
+                               "file://%s"%dest_file],
+                              "Checking xrootd copy from Authenticated cache", user=True)
+        origin_file = os.path.join(getcfg("OriginRootdir"), getcfg("OriginAuthExport").lstrip("/"), name)
+        chechskum_match = files.checksum_files_match(origin_file, dest_file)
+        self.assert_(chechskum_match, 'Cache and download file have same contents')
```
Comment (matyasselmeci) [inline osgtest/tests/test_460_stashcache.py]:
```suggestion
        checksum_match = files.checksum_files_match(origin_file, dest_file)
        self.assert_(checksum_match, 'Origin and file downloaded via cache have the same contents')
```

File: osgtest/tests/test_155_stashcache.py (hunk 1 above)
Comment (efajardo) [reply osgtest/tests/test_155_stashcache.py]: It is the hash of the vdttest user cert.

Comment (matyasselmeci) [reply osgtest/tests/test_155_stashcache.py]: Is it guaranteed to be the same? If so, could you put a constant somewhere or add a comment or something?

Comment (efajardo) [reply osgtest/tests/test_155_stashcache.py]: I added a comment and a constant.

**Pull request merged**